dao-contracts/src/escrow/bidding.rs

```diff
@@ -0,0 +1,272 @@
+use std::{borrow::Borrow, rc::Rc};
+
+use casper_types::{U512, URef};
+use delegate::delegate;
+
+use casper_dao_utils::{
+    Address,
+    BlockTime,
+    casper_dao_macros::Instance,
+    casper_env::{caller, get_block_time},
+    cspr,
+};
+
+use crate::{
+    Configuration,
+    ConfigurationBuilder,
+    escrow::{
+        bid::{Bid, BidStatus, CancelBidRequest, ShortenedBid, SubmitBidRequest},
+        job::{Job, PickBidRequest},
+        job_offer::{CancelJobOfferRequest, JobOffer, PostJobOfferRequest},
+        storage::{BidStorage, JobStorage},
+        types::{BidId, JobOfferId},
+    },
+    refs::{ContractRefs, ContractRefsWithKycStorage},
+    ReputationContractInterface,
+    voting::{kyc_info::KycInfo, onboarding_info::OnboardingInfo},
+};
+
+#[derive(Instance)]
+pub struct Bidding {
+    #[scoped = "contract"]
+    pub bid_storage: BidStorage,
+    #[scoped = "contract"]
+    pub job_storage: JobStorage,
+    #[scoped = "contract"]
+    kyc: KycInfo,
+    #[scoped = "contract"]
+    onboarding: OnboardingInfo,
+    #[scoped = "contract"]
+    refs: ContractRefsWithKycStorage,
+}
+
+impl Bidding {
+    delegate! {
+        to self.bid_storage {
+            pub fn job_offers_count(&self) -> u32;
+            pub fn bids_count(&self) -> u32;
+            pub fn get_job_offer(&self, job_offer_id: JobOfferId) -> Option<JobOffer>;
+            pub fn get_bid(&self, bid_id: BidId) -> Option<Bid>;
+        }
+    }
```

Comment:
I like that part. Bidding has underlying storage and delegates to it some getters. But I wouldn't say I like the storage module is also used in the top-level module - the contract level as a standalone module. I see two methods of accessing this module. Sometimes you get some data directly from `BidStorage` and sometimes you use `Bidding` as a proxy.

Reply:
Fixed